CapstoneProject/Assets/Scripts/Core/Entity/Abilities/FireBall.cs

```diff
@@ -44,7 +44,8 @@ private void OnTriggerEnter2D(Collider2D col)
             HitCollision();
             if (col.GetComponent<EnemyHealthComponent>())
             {
-                _enemyList.Add(col.GetComponent<EnemyHealthComponent>());
+                if (!_enemyList.Contains(col.GetComponent<EnemyHealthComponent>()))
+                    _enemyList.Add(col.GetComponent<EnemyHealthComponent>());
```

Comment:
you do col.GetComponent<Ene... 3 times here, do it once and save to a local var then use that var for all checks.